shell/platform/embedder/embedder.cc

```diff
@@ -565,6 +565,43 @@ static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
   return surface;
 }
 
+static sk_sp<SkSurface> MakeSkSurfaceFromBackingStore(
+    GrDirectContext* context,
+    const FlutterBackingStoreConfig& config,
+    const FlutterMetalBackingStore* metal) {
+#ifdef SHELL_ENABLE_METAL
+  GrMtlTextureInfo texture_info;
+  texture_info.fTexture.reset(metal->texture.texture);
+
+  GrBackendTexture backend_texture(config.size.width,   //
```

Comment:
Should we be checking that the texture given to us by the embedder has size identical to the one mentioned in the config? I think we should do whatever the OpenGL backend does in case of mismatches.

Reply:
The GL backend doesn't do any checking: https://github.com/flutter/engine/blob/master/shell/platform/embedder/embedder.cc#L493 and https://github.com/flutter/engine/blob/master/shell/platform/embedder/embedder.cc#L449

Reply:
Ok, thats fine then. If the sizes are different, the contents will just be clipped.